xdsl/dialects/irdl/irdl.py

```diff
@@ -379,6 +394,64 @@ def print(self, printer: Printer) -> None:
         printer.print(")")
 
 
+def _parse_attribute(parser: Parser) -> tuple[str, SSAValue]:
+    key = parser.parse_str_literal()
+    parser.parse_punctuation("=")
+    arg = parser.parse_operand()
+
+    return (key, arg)
+
+
+@irdl_op_definition
+class AttributesOp(IRDLOperation):
+    """Define the attributes of an operation"""
+
+    name = "irdl.attributes"
+
+    attribute_values = var_operand_def(AttributeType())
+
+    attribute_value_names = attr_def(ArrayAttr[StringAttr])
+
+    def __init__(self, attributes: dict[str, SSAValue]):
+        operands = tuple(attributes.values())
+        names = ArrayAttr(StringAttr(x) for x in attributes.keys())
+        super().__init__(
+            operands=(operands,), attributes={"attribute_value_names": names}
+        )
```

Comment:
I would recommend adding a dumb init with the fields of the op, and a `.get` helper with the smarts, similarly below, there's currently no way to create these ops with exactly the things you want

Reply:
Is there example of such a `get` helper somewhere?

Reply:
Not sure, I just mean a class/static method that returns the Operation